### PR TITLE
conf.h is now used to compile user-ta libraries

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**

### CODE REVIEW

File: documentation/build_system.md [hunk 1]
```diff
@@ -221,7 +221,7 @@ may be defined in `sub.mk` when then pertain to a specific library for instance.
 
 Variables with the `CFG_` prefix are treated in a special
 way: their value is automatically reflected in the generated header
-file `$(out-dir)/core/include/generated/conf.h`, after all the included
+file `$(out-dir)/include/generated/conf.h`, after all the included
 makefiles have been processed. `conf.h` is automatically included by the
 preprocessor when a source file belonging to the TEE core is built.
```
Comment (jforissier) [inline documentation/build_system.md]: `s/belonging to the TEE core//`

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

I noticed that the header in `conf.mk` has:
```
# TEE version (undefined)
```
...so, while you're at it you may want to do:
```
diff --git a/mk/checkconf.mk b/mk/checkconf.mk
index d71c0f6..143b1bd 100644
--- a/mk/checkconf.mk
+++ b/mk/checkconf.mk
@@ -34,7 +34,7 @@ define check-conf-mk
                $(strip $(var)=$($(var))_nl_)))"; \
        mkdir -p $(dir $@);                                     \
        echo "# auto-generated TEE configuration file" >$@.tmp; \
-       echo "# TEE version $${CFG_TEE_VERSION:-(undefined)}" >>$@.tmp; \
+       echo "# TEE version ${CFG_TEE_IMPL_VERSION}" >>$@.tmp; \
        echo "ARCH=${ARCH}" >>$@.tmp;                           \
        echo "PLATFORM=${PLATFORM}" >>$@.tmp;                   \
        echo "PLATFORM_FLAVOR=${PLATFORM_FLAVOR}" >>$@.tmp;     \
```

**Comment by pascal-brand-st-dev** [comment]
Updated. https://github.com/OP-TEE/optee_test/pull/64 should be merged first.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**